src/content/en/updates/2018/01/paintapi.md

```diff
@@ -0,0 +1,278 @@
+{# wf_blink_components: Blink>CSS #}
+
+
+# CSS Paint API {: .page-title }
+
+{% include "web/_shared/contributors/surma.html" %}
+
+## New possibilities in Chrome 65
+CSS Paint API (also known as “CSS Custom Paint” or “Houdini’s Paint Worklet”) is
+about to be enabled by default in Chrome Stable. What is it? What can you do
+with it? And how does it work? Well, read on, will ya’…
+
+
+CSS Paint API allows you to programmatically generate an image whenever a CSS
+property expects an image. Properties like `background-image` or `border-image`
+are usually used with `url()` to load an image file or with CSS built-in
+functions like `linear-gradient()`. Instead of using those, you can now use
+`paint(myPainter)` to reference a _paint worklet_.
+
+### Writing a paint worklet
+
+To define a paint worklet called `myPainter`, we need to load a CSS paint
+worklet file using `CSS.paintWorklet.addModule('my-paint-worklet.js')`. In that
+file we can use the `registerPaint` function to register a paint worklet class:
+
+```js
+class MyPainter {
+  paint(ctx, geometry, properties) {
+    // ...
+  }
+}
+
+registerPaint('myPainter', MyPainter);
+```
+
+Inside the `paint()` callback, we can use `ctx` the same way we would a
+CanvasRenderingContext2D as we know it from `<canvas>`. If you know how to draw
+in a `<canvas>`, you can draw in a paint worklet! `geometry` tells us the width
+and the height of the canvas that is at our disposal. `properties` I will
+explain later in this article.
+
+Note: A paint worklet’s context is not 100% the same as a `<canvas>` context. As
+of now, text rendering methods are missing and for security reasons you cannot
+read back pixels from the canvas.
+
+As an introductory example, let’s write a checkerboard paint worklet and use it
+as a background image of a `<textarea>` (I am using a textarea because it’s
+resizable by default):
+
+```
+<!-- index.html -->
+<!doctype html>
+<style>
+  textarea {
+    background-image: paint(checkerboard);
+  }
+</style>
+<textarea></textarea>
+<script>
+  CSS.paintWorklet.addModule('checkerboard.js');
+</script>
+```
+
+```
+// checkerboard.js
+class CheckerboardPainter {
+  paint(ctx, geom, properties) {
+    // Use `ctx` as if it was a normal canvas
+    const colors = ['red', 'green', 'blue'];
+    const size = 32;
+    for(let y = 0; y < geom.height/size; y++) {
+      for(let x = 0; x < geom.width/size; x++) {
+        const color = colors[(x + y) % colors.length];
+        ctx.beginPath();
+        ctx.fillStyle = color;
+        ctx.rect(x * size, y * size, size, size);
+        ctx.fill();
+      }
+    }
+  }
+}
+
+// Register our class under a specific name
+registerPaint('checkerboard', CheckerboardPainter);
+```
+
+If you’ve used `<canvas>` in the past, this code should look very familiar. See
+the live
+[demo](https://googlechromelabs.github.io/houdini-samples/paint-worklet/checkerboard/)
```

Comment:
this demo crashes canary for me :(

Can you add feature detection to it that displays some explainer text when users hit it in an unsupported browser?

Reply:
Demo works for me? Hopefully it’s just a daily Canary glitch. Good call on the feature detection, will add.